kernel/fatal.c

```diff
@@ -122,7 +124,9 @@ void z_fatal_error(unsigned int reason, const z_arch_esf_t *esf)
 	LOG_ERR("Current thread: %p (%s)", thread,
 		log_strdup(thread_name_get(thread)));
 
+#ifndef CONFIG_XTENSA
```

Comment:
Not worthy of a -1 from me, but CONFIG_XTENSA here might be better with a CONFIG_NO_COREDUMP or similarly named feature disable option rather than CONFIG_XTENSA

Reply:
I #ifndef out this call to coredump with CONFIG_XTENSA here since I need the coredump to run in a different place for Xtensa than on the other architectures (i.e. inside of arch/xtensa/core/fatal.c instead of kernel/fatal.c). We have some #ifdefs with CONFIG_DEBUG_COREDUMP_... to turn the whole the feature on and off elsewhere.

Reply:
That makes much more sense in that context, ignore previous commentary